src/contract_info.rs

```diff
@@ -91,70 +33,15 @@ pub struct ContractInfoV3 {
-    fn from(contract_info: ContractInfo) -> Self {
-        ContractInfoV3 {
-            name: contract_info.name,
-            bind_name: contract_info.bind_name,
-            base_denom: contract_info.base_denom,
-            convertible_base_denoms: contract_info.convertible_base_denoms,
-            supported_quote_denoms: contract_info.supported_quote_denoms,
-            approvers: vec![],
-            executors: contract_info.executors,
-            ask_fee_info: None,
-            bid_fee_info: None,
-            ask_required_attributes: contract_info.ask_required_attributes,
-            bid_required_attributes: contract_info.bid_required_attributes,
-            price_precision: contract_info.price_precision,
-            size_increment: contract_info.size_increment,
-        }
-    }
-}
-
-#[allow(deprecated)]
-impl From<ContractInfoV1> for ContractInfoV3 {
-    fn from(contract_info: ContractInfoV1) -> Self {
-        ContractInfoV3 {
-            name: contract_info.name,
-            bind_name: contract_info.bind_name,
-            base_denom: contract_info.base_denom,
-            convertible_base_denoms: contract_info.convertible_base_denoms,
-            supported_quote_denoms: contract_info.supported_quote_denoms,
-            approvers: contract_info.approvers,
-            executors: contract_info.executors,
-            ask_fee_info: None,
-            bid_fee_info: None,
-            ask_required_attributes: contract_info.ask_required_attributes,
-            bid_required_attributes: contract_info.bid_required_attributes,
-            price_precision: contract_info.price_precision,
-            size_increment: contract_info.size_increment,
-        }
-    }
-}
-
-#[allow(deprecated)]
-impl From<ContractInfoV2> for ContractInfoV3 {
-    fn from(contract_info: ContractInfoV2) -> Self {
-        ContractInfoV3 {
-            name: contract_info.name,
-            bind_name: contract_info.bind_name,
-            base_denom: contract_info.base_denom,
-            convertible_base_denoms: contract_info.convertible_base_denoms,
-            supported_quote_denoms: contract_info.supported_quote_denoms,
-            approvers: contract_info.approvers,
-            executors: contract_info.executors,
-            ask_fee_info: match (contract_info.fee_account, contract_info.fee_rate) {
-                (Some(account), Some(rate)) => Some(FeeInfo { account, rate }),
-                (_, _) => None,
-            },
-            bid_fee_info: None,
-            ask_required_attributes: contract_info.ask_required_attributes,
-            bid_required_attributes: contract_info.bid_required_attributes,
-            price_precision: contract_info.price_precision,
-            size_increment: contract_info.size_increment,
-        }
+/// Checks and enforces the minimum prior contract version supported for migration.
+fn check_minimum_supported_version(current_version: &Version) -> Result<(), ContractError> {
```

Comment:
Ya, so i was basically thinking of having this check in each `migrate_asks`, and `migrate_bids`